.pre-commit-config.yaml

```diff
@@ -0,0 +1,5 @@
+repos:
+- repo: https://github.com/BlankSpruce/gersemi
```

Comment:
Also, is there something guarding me from force pushing a wrongly formatted cmake? Can we make a CI job for it (not necessarily in this PR)

Reply:
Probably easiest with this: https://github.com/pre-commit/action
Then the GitHub action will do the same thing as what runs locally, but catches pushes from a client without pre-commit configured and those that pushed with `--no-verify`.  It'll run slower since it'll check all files, I think.  Maybe some extra configuration to compare against `main` and find a subset to run against.  But if it's fast enough that extra legwork may not be warranted.

Reply:
Lets add pre-commit in CI in a different PR

Reply:
Issue filed.